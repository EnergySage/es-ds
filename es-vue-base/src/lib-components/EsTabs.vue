<template>
    <b-tabs
        class="es-tabs"
        v-bind="$attrs"
        :active-nav-item-class="['font-weight-bolder', 'active-tab']"
        :vertical="vertical"
        :style="cssProps"
        no-nav-style
        v-on="$listeners">
        <slot />
    </b-tabs>
</template>

<script lang="js">
import { BTabs } from 'bootstrap-vue';

export default {
    name: 'EsTabs',
    components: {
        BTabs,
    },
    props: {
        /**
         * Vertical
         * optional, defaults to false
         */
        vertical: {
            type: Boolean,
            default: false,
        },
        /**
         * Border
         */
        border: {
            type: Boolean,
            default: true,
        },
    },
    computed: {
        cssProps() {
            return (
                this.border
                    ? { '--border-bottom': '1px solid' }
                    : { '--border-bottom': '0' }
            );
        },
    },
};
</script>

<style lang="scss">
@import '~@energysage/es-bs-base/scss/includes';

.es-tabs {
    .nav {
        border-bottom: var(--border-bottom);
        border-color: $gray-300;
    }

    .nav-link {
        color: $gray-900;
        font-weight: $font-weight-normal;
        margin: 0 2rem -2px;
        padding: 0.25rem 0 0;

        &:first-child {
            margin-left: 0;
        }

        &:hover,
        &:focus {
            color: $primary;
        }
    }

    .active-tab {
        border-bottom: 2px solid;
        border-color: $primary;
        color: $primary;

        &:hover,
        &:focus {
            color: $primary;
        }
    }
}
</style>
