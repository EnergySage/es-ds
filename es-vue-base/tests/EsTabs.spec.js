import { mount } from '@vue/test-utils';
import EsTabs from '@/src/lib-components/molecules/EsTabs.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsTabs', () => {
    // Basic test; should exist for most components
    test('<EsTabs />', async () => {
        const wrapper = mount(EsTabs, {
            ...jestVue,
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });

    // Test border prop is there
    test('<EsTabs border=false />', async () => {
        const wrapper = mount(EsTabs, {
            ...jestVue,
            propsData: {
                border: false,
            },
        });
        expect(wrapper.props('border')).toBe(false);
        expect(wrapper.html()).toMatchSnapshot();
    });

    // Test slot is there
    test('<EsTabs>My Content</EsTabs>', async () => {
        const wrapper = mount(EsTabs, {
            ...jestVue,
            slots: {
                default: 'My Content',
            },
        });

        expect(wrapper.vm.$slots.default[0].text).toBe('My Content');
        expect(wrapper.html()).toMatchSnapshot();
    });
});
