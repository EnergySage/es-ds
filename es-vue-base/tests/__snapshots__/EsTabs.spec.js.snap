// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsTabs <EsTabs /> 1`] = `
"<div class=\\"tabs\\" style=\\"--border-bottom: 0; --border-color: null;\\" id=\\"__BVID__4\\">
  <!---->
  <div class=\\"\\">
    <ul role=\\"tablist\\" class=\\"nav\\" id=\\"__BVID__4__BV_tab_controls_\\">
      <!---->
      <!---->
    </ul>
  </div>
  <div class=\\"tab-content\\" id=\\"__BVID__4__BV_tab_container_\\">
    <div class=\\"tab-pane active\\"></div>
  </div>
</div>"
`;

exports[`EsTabs <EsTabs border=false /> 1`] = `
"<div class=\\"tabs\\" style=\\"--border-bottom: 0; --border-color: null;\\">
  <!---->
  <div class=\\"\\">
    <ul role=\\"tablist\\" class=\\"nav\\">
      <!---->
      <!---->
    </ul>
  </div>
  <div class=\\"tab-content\\">
    <div class=\\"tab-pane active\\"></div>
  </div>
</div>"
`;

exports[`EsTabs <EsTabs>My Content</EsTabs> 1`] = `
"<div class=\\"tabs\\" style=\\"--border-bottom: 0; --border-color: null;\\">
  <!---->
  <div class=\\"\\">
    <ul role=\\"tablist\\" class=\\"nav\\">
      <!---->
      <!---->
    </ul>
  </div>
  <div class=\\"tab-content\\">My Content
    <!---->
  </div>
</div>"
`;
